lib/Semantics/resolve-names.cpp

```diff
@@ -1548,26 +1554,70 @@ bool AttrsVisitor::SetBindNameOn(Symbol &symbol) {
 
 void AttrsVisitor::Post(const parser::LanguageBindingSpec &x) {
   CHECK(attrs_);
-  attrs_->set(Attr::BIND_C);
-  if (x.v) {
-    bindName_ = EvaluateExpr(*x.v);
+  if (attrs_->test(Attr::BIND_C)) {
+    Say(currStmtSource().value(),
+        "Attribute 'BIND(C)' cannot be used more than once"_err_en_US);
+  } else {
+    attrs_->set(Attr::BIND_C);
+    if (x.v) {
+      bindName_ = EvaluateExpr(*x.v);
+    }
   }
 }
 bool AttrsVisitor::Pre(const parser::IntentSpec &x) {
   CHECK(attrs_);
-  attrs_->set(IntentSpecToAttr(x));
+  CheckAndSet(IntentSpecToAttr(x));
   return false;
 }
 bool AttrsVisitor::Pre(const parser::Pass &x) {
-  if (x.v) {
-    passName_ = x.v->source;
-    MakePlaceholder(*x.v, MiscDetails::Kind::PassName);
-  } else {
-    attrs_->set(Attr::PASS);
+  if (CheckAndSet(Attr::PASS)) {
+    if (x.v) {
+      passName_ = x.v->source;
+      MakePlaceholder(*x.v, MiscDetails::Kind::PassName);
+    }
   }
   return false;
 }
 
+// C730, C743, C755, C778, C1543 say no attribute or prefix repetitions
+bool AttrsVisitor::IsDuplicateAttr(Attr attrName) {
+  if (attrs_->test(attrName)) {
+    Say(currStmtSource().value(),
+        "Attribute '%s' cannot be used more than once"_err_en_US,
```

Comment:
Why a fatal error?  It's not going to hurt anything.

Reply:
Thanks for pointing this out.  I agree with you.  I just checked, and duplicate attributes don't seem to be noticed by pgf90, but they're fatal errors for both IBM and Intel.

I'll change the duplicates to warnings.

Reply:
Actually, some attributes such as CODIMENSION, EXTENDS, DIMENSION convey extra information.  It seems appropriate to me that duplicating these attributes should be an error.

Reply:
EXTENDS is an optional part of the definition of a derived type, not an attribute of an entity.

I'm not saying that we should allow all possible duplicated attributes (besides the ones that the standard says we have to allow).